js/main.js

```diff
@@ -118,19 +158,45 @@ function goBack(event) {
   $viewPage.className = 'viewpage hidden';
   $weatherPage.className = 'weatherpage hidden';
   $headerColor.className = 'header normal';
-  removeWeatherInfo($weatherContainer);
-  removeWeatherInfo($weatherImageContainer);
+  $viewPageHeader.className = 'hidden';
+  removeContainer($weatherContainer);
+  removeContainer($weatherImageContainer);
+  removeContainer($viewImageContainer);
+  removeContainer($viewFullContainer);
+  removeContainer($errorContainer);
+  $viewFullContainer.remove();
   $mainHeading.className = 'heading view';
   $weatherHeading.className = 'hidden';
   changeBackground('snow');
 }
 
-function removeWeatherInfo(parent) {
+function removeContainer(parent) {
   while (parent.firstChild) {
     parent.removeChild(parent.firstChild);
   }
 }
 
+function deleteFromStorage(event) {
+  var $idNum = Number(idNum);
+  var $storageForRemoval = localStorage.getItem('painting-storage');
+  var $parsed = JSON.parse($storageForRemoval);
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].id === $idNum) {
+      data.entries.splice(i, 1);
+      if ($parsed !== null) {
+        $parsed.entries.splice(i, 1);
+        window.addEventListener('beforeunload', function () {
+          localStorage.setItem('painting-storage', JSON.stringify($parsed));
+        });
+      }
+      // data.nextEntryId--;
+    }
+  }
+  removeContainer($viewImageContainer);
+  removeContainer($viewFullContainer);
+  $viewFullContainer.remove();
+  goToView();
+}
```

Comment:
Rather than deleting from localStorage and saving to localStorage amongst your code here, you should move that to a different file named `data.js`, like we've done previously.